### ISSUE TITLE
enumeration timeout with linux 4.13

### ISSUE THREAD

**Issue by BlackIkeEagle** [dkopecek/usbguard#204]
ERROR: UEventDeviceManager: present devices: enumeration timeout

Could this have something to do with the kernel config?

https://github.com/herecura/linux-bede/blob/herecura/config-desktop.x86_64

**Comment by dkopecek** [comment]
Hi and thanks for the bug report. I'm aware of this and working on a fix (different device enumeration procedure) It should be available in the 0.7.1 release.

**Comment by BlackIkeEagle** [comment]
superb, thanks.

**Comment by dkopecek** [comment]
@BlackIkeEagle I'm working on a fix in the private-kernel-4.13 branch. Would you have time to checkout the branch and check whether it fixes the problem for you? What distro are you using? I can provide testing builds for Fedora/CentOS/RHEL distros.

**Comment by BlackIkeEagle** [comment]
thanks @dkopecek , I'll test later, I'm on Arch Linux

**Comment by BlackIkeEagle** [comment]
@dkopecek first test: applying the patch on top of 0.7.0 fixes `usbguard generate-policy` while using linux 4.13.1

**Comment by BlackIkeEagle** [comment]
@dkopecek and for completeness I have also tested the branch and it also works there for me

**Comment by dkopecek** [comment]
@BlackIkeEagle Thanks a lot for testing!

**Comment by BlackIkeEagle** [comment]
@dkopecek as a bonus, just tested with linux 4.9.49 and the behaviour is the same

**Comment by genodeftest** [comment]
Thanks for the fix!

Any chance this bug fix can land in a bugfix release? I just ran into the same issue on kernel 4.13.4, which has landed in Fedora's updates-testing repo.

On Fedora, I had to change permissions on /etc/usbguard/usbguard-daemon.conf and /etc/usbguard/rules.conf using `chmod og-r` before usbguard could run again. I'm using the builds from https://copr.fedorainfracloud.org/coprs/mildew/usbguard-testing/.
